src/components/application_manager/src/rpc_handler_impl.cc

```diff
@@ -324,6 +375,8 @@ bool RPCHandlerImpl::ConvertMessageToSO(
                     "Convertion result: "
                         << result << " function id "
                         << output[jhs::S_PARAMS][jhs::S_FUNCTION_ID].asInt());
+      // TODO Figure out whether to use the RemoveUnknownParams bool for hmi
```

Comment:
Shouldn't need to leave unknown parameters for HMI services. Unknown parameter passing is meant for interactions between mobile app services so that it can be independent from the version of the head unit.

Reply:
The filtering of UnknownParameters for HMI requests was set to false by default(not in my changes). I was wondering if there was some reason for that. 

Also since were are allowing unknown parameters for app service RPCs that would include RPCs on the HMI side(ASPublishAppService, ASGetAppServiceData, OSCU notification)

Reply:
I don't have an issue with keeping unknown parameters in these cases, I just wouldn't suggest going out your way to make it one way or another. Just remove the TODO.

Reply:
Fixed 👍